### DISCUSSION TITLE
Can I have a mixture of connect-go and protoc-go running at the same time?

### DISCUSSION THREAD

**Discussion by user3301** [connectrpc/connect-go#501] [Q&A] [answered]
Hi, 
Just wondering if I have both `protoc` generated Go server code as well as `connect-go` generated Go server code, am I unable to run both side-by-side? Something like:
```go
grpcServer := grpc.NewServer()

mux := http.NewServerMux()

// protoc service
pb.RegisterNormalGRPCService1(grpcServer, myService)

// connect-go
path, handler := connect.NewBufAPIHander(myService)
mux.Handle(path, handler)

http.Server {
   Handler : h2c.NewHandler(mux, &http2.Server{})
}.ListenAndServe()
```
If I setup my server like this and run, I can hit the connect-go endpoint okay but when I use protoc client to send gRPC requests 
```go
NormalGRPCServiceClient.MyAPI(request)
```
I got 
```sh
rpc error: code = Unimplemented desc = unexpected HTTP status code received from server: 404 (Not Found); transport: received unexpected content-type "text/plain; charset=utf-8"
```

**Comment by akshayjshah** [comment, marked as answer]
Hi! Sorry for the late reply, @user3301.

You can have both `grpc-go` and `connect-go` running side-by-side, but typically they'll be on separate ports. In your pseudocode above, you never call `grpcServer.Serve`.

Of course, `connect-go` supports the gRPC protocol, so you don't need `grpc-go` :)
